### PR TITLE
GH-107463: Moved most extension modules to the pythoncore project file

### PR DESCRIPTION
**Pull request by AraHaan**
For this I had to start with making the changes to the 3.11 branch, then a few additional changes in the 3.12 branch, before finally adding some minor fixups in the main branch due to the following:
- 3.12 added the ``_wmi`` c++ extension module (C++ 20 standard)
- 3.13 removed the ``_msi`` c extension module and project file (which means needing to remove it from ``config.c`` and from ``pythoncore.vcxproj`` when I open pr to ``main``)

Note: I lack the ability to edit the build scripts for linux and macos building so someone would have to assist with bringing these changes to those scripts as well.

Also I would recommend making changes starting with whichever one is easier (the 3.11 version, the 3.12 version, or this version from main (3.13)).

* Issue: gh-107463

### CODE REVIEW

File: PCbuild/bz2.vcxproj [hunk 1]
```diff
@@ -67,31 +67,33 @@
...
     <RootNamespace>bz2</RootNamespace>
     <Keyword>Win32Proj</Keyword>
   </PropertyGroup>
   <Import Project="python.props" />
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.Default.props" />
   <PropertyGroup Label="Configuration">
-    <ConfigurationType>DynamicLibrary</ConfigurationType>
+    <OutDir>$(BuildPath)</OutDir>
+    <IntDir>$(IntDir.Replace(`\\`, `\`))</IntDir>
+    <TargetName>$(TargetName)$(PyDebugExt)</TargetName>
```
Comment (AraHaan) [inline PCbuild/bz2.vcxproj]:
```suggestion
```

File: PCbuild/libmpdec.vcxproj [hunk 2]
```diff
@@ -67,38 +67,41 @@
...
+    <RootNamespace>libmpdec</RootNamespace>
     <Keyword>Win32Proj</Keyword>
   </PropertyGroup>
   <Import Project="python.props" />
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.Default.props" />
   <PropertyGroup Label="Configuration">
-    <ConfigurationType>DynamicLibrary</ConfigurationType>
+    <OutDir>$(BuildPath)</OutDir>
+    <IntDir>$(IntDir.Replace(`\\`, `\`))</IntDir>
+    <TargetName>$(TargetName)$(PyDebugExt)</TargetName>
```
Comment (AraHaan) [inline PCbuild/libmpdec.vcxproj]:
```suggestion
```

File: PC/dl_nt.c [hunk 3]
```diff
@@ -24,9 +24,13 @@ BOOL    WINAPI  DllMain (HANDLE hInst,
     switch (ul_reason_for_call)
     {
         case DLL_PROCESS_ATTACH:
+        {
             PyWin_DLLhModule = hInst;
-            break;
 
+            // This should be safe to call here?
+            DisableThreadLibraryCalls(hInst);
```
Comment (AraHaan) [inline PC/dl_nt.c:31]: Is it ok to call ``DisableThreadLibraryCalls(hInst);`` here?

**Comment by AA-Turner** [comment]
I'm going to close this PR until there's consensus for this change on the linked issue (#107463). Feel free to continue experimenting on your fork!

A

**Pull request closed without merging**